CONTRIBUTING.md

```diff
@@ -36,61 +34,210 @@ $ sudo dnf install device-mapper-devel btrfs-progs-devel
+tabs on what the broker is doing, running the broker locally is an excellent idea.
+
+Before attempting to run the broker locally, you will need to have deployed an
+[OpenShift Origin Cluster with the Ansible Service Broker](#deploy_an_openshift_origin_cluster_with_the_ansible_service_broker).
+You can verify this by running `oc get all --all-namespaces` and looking for the `ansible-service-broker`
+`service-catalog` running in the cluster.
+
+**Configuration**
+
+Next, you will need configure your local development variables:
+
+```
+cp scripts/my_local_dev_vars.example scripts/my_local_dev_vars
+```
+
+Now you can modify `scripts/my_local_dev_vars` with things like your `DOCKERHUB_USERNAME`
+or use an insecure broker with `BROKER_INSECURE="true"`.
 
-**Config**
-A broker is configured via the `config.yaml` file. It's recommended to
-copy over `etc/example-config.yaml` to `etc/ansible-service-broker/config.yaml`, and edit
-as desired.
+**Prepare Local Environment**
 
-See the [Broker Configuration](docs/config.md) doc for other example
-configurations.
+Running `make prepare-local-env` will do several things on your behalf:
+
+* Remove the running `ansible-service-broker` from the cluster, leaving only
+  etcd running in the namespace.
+* Modify the `asb` service to point where our locally running broker will be.
+* Generate a configuration file for the broker (`etc/generated_local_development.yaml`)
+
+This would be a good time to have a look at the broker's configuration and make
+changes. Have a look at [the broker configuration examples](docs/config.md).
+By default, the registry section of the config will look something like:
+
+```
+registry:
+  - type: dockerhub
+    name: dh
+    url: https://registry.hub.docker.com
+    user: changeme
+    pass: changeme
+    org: ansibleplaybookbundle
+```
+
+It may server your purposes to append additional registries, for example, if
+you wanted to develop APB's in your organization while still seeing those in
+the ansibleplaybookbundle organization. That may look something like:
+
+```
+registry:
+  - type: dockerhub
+    name: dh
+    url: https://registry.hub.docker.com
+    user: changeme
+    pass: changeme
+    org: ansibleplaybookbundle
+  - type: dockerhub
+    name: example
+    url: https://registry.hub.docker.com
+    user: changeme
+    pass: changeme
+    org: example
+```
+
+**Start the Broker**
+
+With the environment prepared, running your broker is as simple as running
+`make run`.
+
+## Deploy Broker From Image
+
+Once you have [built a Docker image with your broker binary](#package-your-broker-using-docker),
+you can run your broker inside the cluster. Now is an excellent time to point out that
+[fusor/catasb](https://github.com/fusor/catasb) gives you the ability to deploy
+an OpenShift Origin Cluster with the Service Catalog and **your** broker image. If that
+still feels like overkill you can use `make deploy` to replace the running broker with
+one of your choosing.
```

Comment:
The "Once you have build a Docker image..." is confusing when it follows immediately after the make run line. It feels like I have to now build an image even though I was just told how to run the broker locally. Some how we need to distinguish that this section is ONLY if they didn't choose to run locally.

Maybe something like the following:

CHANGE: "Once you have [built a Docker image with your broker binary](#package-your-broker-using-docker), you can run your broker inside the cluster. "
->
"If you chose to [build a Docker image with your broker binary](#package-your-broker-using-docker), you can run your broker inside the cluster. "